code/obj/item/tool/hedron_beam.dm

```diff
@@ -0,0 +1,60 @@
+TYPEINFO(/obj/item/mining_tool/hedron_beam)
+	mats = list("MET-2"=15, "CON-1"=8, "claretine"=10, "koshmarite"=2 )
+
+/obj/item/mining_tool/hedron_beam
+	name = "\improper Hedron beam device"
+	desc = "A prototype multifunctional industrial tool capable of rapidly switching between welding and mining modes."
+	icon = 'icons/obj/items/mining.dmi'
+	icon_state = "hedron-W"
+	inhand_image_icon = 'icons/mob/inhand/hand_guns.dmi'
+	item_state = "gun"
+	c_flags = ONBELT
+	tool_flags = TOOL_WELDING
+	force = 10
+	dig_strength = 3
+	digcost = 2
+	cell_type = /obj/item/ammo/power_cell
+	hitsound_charged = 'sound/items/Welder.ogg'
+	hitsound_uncharged = 'sound/impact_sounds/Metal_Clang_3.ogg'
+
+	proc/mode_toggle()
+		if(src.status)
+			set_icon_state("hedron-W")
+			flick(("hedron-MtoW"), src)
+			src.power_down()
+		else
+			set_icon_state("hedron-M")
+			flick(("hedron-WtoM"), src)
+			src.dig_strength = 3
+			src.power_up()
+
+	power_down() //separate for power depletion power-down
+		src.dig_strength = 0
+		..()
+
+	attack_self(var/mob/user as mob)
+		if (!digcost)
```

Comment:
Not sure what this is doing here (I assume copy paste). Is there any reasonable circumstance where the dig cost becomes 0???

Reply:
Tidied up in 06b6541.